homeassistant/components/eafm/station.py

```diff
@@ -0,0 +1,102 @@
+"""Support for a monitoring station."""
+from datetime import timedelta
+import logging
+
+from aioeafm import get_station
+import aiohttp
+
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.event import async_track_time_interval
+
+from .sensor import Measurement
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=15)
+
+
+class Station:
+    """A collection of gauges at a flood monitoring station."""
+
+    def __init__(self, hass, entry):
+        """Initialise the station."""
+        self.hass = hass
+        self.entry = entry
+        self.station = entry.data["station"]
+        self.measurements = {}
+        self.available = True
+        self._poller_remover = None
+        self._async_add_entities = None
+
+    @property
+    def is_polling(self):
+        """Is the station being actively monitored."""
+        return self._poller_remover is not None
+
+    async def async_start(self):
+        """Start polling the API."""
+        self._poller_remover = async_track_time_interval(
```

Comment:
Please use our update coordinator helper instead.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
Done.